uber/site_sections/registration.py

```diff
@@ -318,12 +319,18 @@ def check_in(self, session, id, age_group, group, badge_num=None, message=''):
                     message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
 
             if group:
-                session.match_to_group(attendee, session.group(group))
+                g = sesson.group(group)
+                if g.amount_unpaid:
+                    message = 'That group has an outstanding balance of ${}'.format(g.amount_unpaid)
+                else:
+                    session.match_to_group(attendee, g)
```

Comment:
I saw the group leaders recently got emails about assigning their group badges...